generator/OpaqueGen.cs

```diff
@@ -159,6 +159,18 @@ public override void Generate (GenerationInfo gen_info)
 			}
 
 			if (finalizer_needed) {
+				// Generate Dispose override.
+				sw.WriteLine ("\t\tpublic override void Dispose ()");
+				sw.WriteLine ("\t\t{");
+				sw.WriteLine ("\t\t\tif (Owned)");
+				if (dispose != null)
+					sw.WriteLine ("\t\t\t\t{0} (Raw);", dispose.CName);
+				else if (unref != null)
+					sw.WriteLine ("\t\t\t\t{0} (Raw);", unref.CName);
+				sw.WriteLine ("\t\t\tbase.Dispose();");
+				sw.WriteLine ("\t\t}");

```

Comment:
I presume this will crash gloriously if we double-dispose unless we null out 'Raw' after the unref?